pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -248,6 +254,20 @@ func (d *Driver) Start() error {
 		return err
 	}
 
+	// when loading a VM created by a crc version predating this commit,
+	// d.VsockQemuGAPort will be missing from ~/.crc/machines/crc/config.json
+	// In such a case, assume the VM will not support time sync
+	if d.VsockQemuGAPort != 0 {
```

Comment:
makes sense...

perhaps something that can also be detected in the main binary ?

Reply:
I prefer to detect it there, having a workaround for crc's behaviour in vfkit's code would be odd.

Reply:
The problem is that you can't report back from the driver that this isn't possible.
We would like to inform the user about this error in some form... or we assume this will never happen?

Reply:
It's something specific to the machine driver. With an older crc, you run `crc start; crc stop`, the machine driver state is serialized to json, and does not contain a `VsockQemuGAPort` field because this did not exist when it was built.
Then `crc start` is ran with a new crc build which looks for `VsockQemuGAPort` in the serialized json. It will get 0 instead of the expected value since the field was not present.